### PR TITLE
Add parameter to set revision `--revision` for the Azure Service Mesh addon while creating AKS cluster.

### PR DESCRIPTION
**Pull request by deveshdama**
This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command
`az aks create -g myResourceGroup -n myCluster --enable-azure-service-mesh --revision asm-1-17`

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

File: src/aks-preview/azext_aks_preview/tests/latest/test_managed_cluster_decorator.py [hunk 1]
```diff
@@ -4867,7 +4868,9 @@ def test_set_up_azure_service_mesh(self):
         ground_truth_mc_2 = self.models.ManagedCluster(
             location="test_location",
             service_mesh_profile=self.models.ServiceMeshProfile(
-                mode="Istio", istio=self.models.IstioServiceMesh()
+                mode="Istio", istio=self.models.IstioServiceMesh(
+                    revisions=["asm-1-88"]
```
Comment (SanyaKochhar) [inline src/aks-preview/azext_aks_preview/tests/latest/test_managed_cluster_decorator.py:4872]: do we also need a test case here for if revisions is empty? or is that implicit?

Comment (deveshdama) [reply src/aks-preview/azext_aks_preview/tests/latest/test_managed_cluster_decorator.py:4872]: thanks for pointing this out. Added a UT for the same.

**Review by FumingZhang — COMMENTED** [review]
lgtm

It would be great if you could help update one of the existing live test cases to cover the new option.

**Review by FumingZhang — COMMENTED** [review]
lgtm

It would be great if you could help update one of the existing live test cases to cover the new option.

File: src/aks-preview/HISTORY.rst [hunk 2]
```diff
@@ -24,6 +24,7 @@ Pending
 * Add `--ssh-access` to the `az aks nodepool add` command.
 * Add `--ssh-access` to the `az aks nodepool update` command.
 * Implicitly enable istio when ingress or egress gateway is enabled for Azure Service Mesh.
+* Add parameter to set revision `--revision` for the Azure Service Mesh addon while creating AKS cluster.
```
Comment (FumingZhang) [inline src/aks-preview/HISTORY.rst:27]: 1.0.0b6/2.0.0b1 has been released, please rebase from main and choose a new version or put the note to pending

File: src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py [hunk 3]
```diff
@@ -63,6 +63,12 @@ def _get_version_in_range(
                 return version
         return ""
 
+    def _get_asm_supported_revision(self):
+        revisions_cmd = "aks mesh get-revisions -l westus2"
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:67]: It would be better to pass the location as a parameter as the test case may not always run in westus2 (e.g., it's set to run in westcentralus in cli runner)

**Review by FumingZhang — APPROVED** [review]
LGTM

File: src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py [hunk 4]
```diff
@@ -63,6 +63,12 @@ def _get_version_in_range(
                 return version
         return ""
 
+    def _get_asm_supported_revision(self, location):
+        revisions_cmd = "aks mesh get-revisions -l {location}"
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:67]:
```suggestion
        revisions_cmd = f"aks mesh get-revisions -l {location}"
```

**Review by zhoxing-ms — APPROVED** [review]

**Comment by zhoxing-ms** [comment]
@wangzelin007 Could you please bypass the CI issue?

**Review by wangzelin007 — APPROVED** [review]
